### PR TITLE
fix(TopBar): migrate to NcPopover

### LINKED ISSUE #13328: Room description pop-up closes to fast
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Create a talk room 
2. Add a  longer description of the room which contains a link at the bottom
3. Hover with the mouse over the room description
4. When the pop-up shows the complete description, try to click the link in the description

### Expected behaviour

Pop-up should stay open while I navigate down to the link, e.g. as long as the cursor is within the pop-up

### Actual behaviour

Popup closes as soon as the mouse moves away from the top area

## Talk app

**Talk app version:** (see apps administration page: `/index.php/settings/apps`)

**Custom Signaling server configured:** yes/no and version (see  Talk administration settings: `/index.php/settings/admin/talk#signaling_server`)

**Custom TURN server configured:** yes/no (see Talk administration settings: `/index.php/settings/admin/talk#turn_server`)

**Custom STUN server configured:** yes/no (see  Talk administration settings: `/index.php/settings/admin/talk#stun_server`)

## Browser

**Microphone available:** yes/no

**Camera available:** yes/no

**Operating system:** Windows/Ubuntu/Mac/...

**Browser name:** Firefox/Chrome/Safari/...

**Browser version:** 124/125/...

### Browser log

<details>
```
Insert your browser log here, this could for example include:
a) The javascript console log
b) The network log
c) ...
```

</details>

## Server configuration

**Operating system**: Ubuntu/RedHat/...

**Web server:** Apache/Nginx

**Database:** MySQL/Maria/SQLite/PostgreSQL

**PHP version:** 8.1/8.2/8.3

**Nextcloud Version:** (see administration page)

**List of activated apps:**

<details>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your server installation folder
```
</details>

**Nextcloud configuration:**

<details>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder
```
</details>

### Server log (data/nextcloud.log)
<details>

```
Insert your server log here
```
</details>

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Fix #13328

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
image | image

### 🏁 Checklist

- [ ] 🌏 Tested with different browsers / clients:
  - [ ] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [ ] Talk Desktop
  - [ ] Not risky to browser differences / client
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/TopBar/TopBar.vue [hunk 1]
```diff
@@ -34,18 +34,20 @@
...
+						:popper-triggers="['hover']"
+						:triggers="['hover']">
+						<template #trigger="{ attrs }">
+							<p v-bind="attrs"
+								class="description"
+								:class="{'description__in-chat' : !isInCall }">
+								{{ conversation.description }}
+							</p>
+						</template>
+						<div class="description__popover" v-html="renderedDescription" />
```
Comment (Antreesy) [inline src/components/TopBar/TopBar.vue:49]: `<!-- eslint-disable-next-line vue/no-v-html -->`

File: src/components/TopBar/TopBar.vue [hunk 2]
```diff
@@ -438,6 +441,10 @@ export default {
 	}
 }
 
+.description__popover {
+	padding: var(--default-grid-baseline);
```
Comment (Antreesy) [inline src/components/TopBar/TopBar.vue:445]: Check with Marco, I remember that we agreed on 50em for text width
```suggestion
	padding: var(--default-grid-baseline);
	width: fit-content;
	max-width: 50em;
```

File: src/components/TopBar/TopBar.vue [hunk 3]
```diff
@@ -34,18 +34,20 @@
...
-								html: true,
-							}"
-							class="description"
-							:class="{'description__in-chat' : !isInCall }">
-							{{ conversation.description }}
-						</p>
-					</template>
+					<NcPopover v-if="conversation.description"
+						:focus-trap="false"
+						:delay="500"
```
Comment (Antreesy) [inline src/components/TopBar/TopBar.vue:39]: would be nice to set `boundary` prop to keep it inside of the chat area

**Review by Antreesy — APPROVED** [review]
🫎

File: src/components/TopBar/TopBar.vue [hunk 4]
```diff
@@ -37,7 +37,9 @@
 					<NcPopover v-if="conversation.description"
 						:focus-trap="false"
 						:delay="500"
+						:boundary="boundaryElement"
 						:popper-triggers="['hover']"
+						:placements="['bottom']"
```
Comment (Antreesy) [inline src/components/TopBar/TopBar.vue:42]: typo? and it's default one, so we can omit it: https://floating-vue.starpad.dev/api/#placement

Comment (DorraJaouad) [reply src/components/TopBar/TopBar.vue:42]: yeah, I didn't check if it is by default

File: src/components/TopBar/TopBar.vue [hunk 5]
```diff
@@ -34,18 +34,22 @@
...
-						</p>
-					</template>
+					<NcPopover v-if="conversation.description"
+						:focus-trap="false"
+						:delay="500"
+						:boundary="boundaryElement"
+						:popper-triggers="['hover']"
+						:triggers="['hover']">
+						<template #trigger="{ attrs }">
+							<p v-bind="attrs"
```
Comment (ShGKme) [inline src/components/TopBar/TopBar.vue:44]: `<p>` cannot have `aria-` attributes from `attr`, it is not valid.

But it is also not valid to make `<p>` an interactive trigger...

Comment (Antreesy) [reply src/components/TopBar/TopBar.vue:44]: it's already inside of `<a>`, which triggers settings dialog to open, and `<p>` only listening to mouseover
So, not much we can do?

File: src/components/TopBar/TopBar.vue [hunk 6]
```diff
@@ -34,18 +34,22 @@
...
-								delay: { show: 500, hide: 500 },
-								autoHide: false,
-								html: true,
-							}"
-							class="description"
-							:class="{'description__in-chat' : !isInCall }">
-							{{ conversation.description }}
-						</p>
-					</template>
+					<NcPopover v-if="conversation.description"
```
Comment (Antreesy) [inline src/components/TopBar/TopBar.vue:37]: Just noticed, that popover might be wider than a description:
image

**Pull request merged**